File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        answer = 0
        i = 0
        j = len(heights)-1
        while i<j:
            if heights[i]<heights[j]:
                tmp = heights[i]*(j-i)
                i += 1
            else:
                tmp = heights[j]*(j-i)
                j -= 1

            if tmp>answer:
                answer = tmp

        return answer

File: Data Structures & Algorithms/daily-temperatures/submission-1.java
class Solution {
  public int[] dailyTemperatures(int[] temperatures) {
    Stack<int[]> st = new Stack<>();
    int[] answer = new int[temperatures.length];
    for (int i = 0; i < temperatures.length; i++) {
      if (st.empty()) {
        st.push(new int[] { temperatures[i], i });
      } else {
        while (!st.empty() && st.peek()[0] < temperatures[i]) {
          answer[st.peek()[1]] = i - st.pop()[1];
        }
        st.push(new int[] { temperatures[i], i });
      }
    }
    return answer;
  }
}